src/managed/Microsoft.Extensions.DependencyModel/Resolution/AppBaseCompilationAssemblyResolver.cs

```diff
@@ -47,7 +47,8 @@ public bool TryResolveAssemblyPaths(CompilationLibrary library, List<string> ass
             var isPackage = string.Equals(library.Type, "package", StringComparison.OrdinalIgnoreCase);
             if (!isProject &&
                 !isPackage &&
-                !string.Equals(library.Type, "referenceassembly", StringComparison.OrdinalIgnoreCase))
+                !string.Equals(library.Type, "referenceassembly", StringComparison.OrdinalIgnoreCase) &&
+                !string.Equals(library.Type, "reference", StringComparison.OrdinalIgnoreCase))
```

Comment:
If you decide against a separate class then the code below this
```if (!isProject && !isPackage && !isPublished)```
will need to change to
```if (library.Type equals “referenceassembly” && !isPublished)```